mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -42,14 +42,17 @@ import org.apache.spark.util.VersionUtils.majorMinorVersion
  * otherwise the features will not be mapped evenly to the columns.
  */
 @Since("1.2.0")
-class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
+class HashingTF @Since("3.0.0") (
+    @Since("1.4.0") override val uid: String,
+    private[ml] val hashFunc: Any => Int)
```

Comment:
This will probably generate a MiMa failure? why does it need to be a constructor param?

Reply:
This also lets people pass arbitrary functions, but is that going to serialize in general? do you want to instead expose a setter that takes some symbolic representation? that is you don't really want to accept arbitrary functions here

Reply:
* I preserve old constructor so mima pass.
* Put it in constructor can make it immutable. Otherwise we have to make it a `var` variable.
> do you want to instead expose a setter that takes some symbolic representation ?

I will update.